ieproxy_windows.go

```diff
@@ -24,8 +24,70 @@ func getConf() ProxyConf {
 }
 
 func writeConf() {
-	regedit, _ := readRegedit()
-	windowsProxyConf = parseRegedit(regedit)
+	var (
+		cfg *tWINHTTP_CURRENT_USER_IE_PROXY_CONFIG
+		err error
+	)
+
+	if cfg, err = getUserConfigFromWindowsSyscall(); err != nil {
+		regedit, _ := readRegedit() //If the syscall fails, backup to manual detection.
+		windowsProxyConf = parseRegedit(regedit)
+		return
+	}
+
+	defer globalFree.Call(uintptr(unsafe.Pointer(cfg.lpszAutoConfigUrl)))
+	defer globalFree.Call(uintptr(unsafe.Pointer(cfg.lpszProxyBypass)))
+	defer globalFree.Call(uintptr(unsafe.Pointer(cfg.lpszProxy)))
```

Comment:
globalFree is not documented to be safe to call with a nil value. And some on these could (in theory I think) be nil.  I suspect it IS safe (enough) to call with nil, but best practice seems to be to create our own wrapper function and defer the call to that.

E.g. defer callGlobalFree(cfg.lpzsAutoConfigUrl)

...
func callGlobalFree(p *unit32){
if p != nil {
globalFree.Call(unitptr(usafe.Pointer(p))
}
}

Reply:
Thanks for catching that. I'll go ahead and throw the wrapper on it when I get in tomorrow.